src/lib/snark_worker/entry.ml

```diff
@@ -196,34 +200,39 @@ let main (module Rpcs_versioned : Intf.Rpcs_versioned_S) ~logger ~proof_level
             in
             log_and_retry "performing work" e (retry_pause 10.) go
         | Ok result ->
-            emit_proof_metrics result.metrics
-              (Selector.Result.Stable.Latest.transactions result)
-              logger ;
+            (* TODO: bring back metrics in subsequent PRs *)
+            (* emit_proof_metrics result.metrics *)
+            (*   (Selector.Result.Stable.Latest.transactions result) *)
+            (*   logger ; *)
             [%log info] "Submitted completed SNARK work $work_ids to $address"
-              ~metadata:
-                [ ("address", `String (Host_and_port.to_string daemon_address))
-                ; ( "work_ids"
-                  , Transaction_snark_work.Statement.compact_json
-                      (One_or_two.map (Work.Spec.instances work)
-                         ~f:Work.Single.Spec.statement ) )
-                ] ;
+              ~metadata:[ address_json; work_ids_json ] ;
             let rec submit_work () =
               match%bind
-                dispatch Rpcs_versioned.Submit_work.Latest.rpc
+                dispatch Rpc_submit_work.Stable.Latest.rpc
                   shutdown_on_disconnect result daemon_address
               with
               | Error e ->
                   log_and_retry "submitting work" e (retry_pause 10.)
                     submit_work
-              | Ok () ->
+              | Ok `Ok ->
+                  go ()
```

Comment:
Suggestion for future PR: use [forever](https://ocaml.janestreet.com/ocaml-core/v0.12/doc/async_kernel/Async_kernel/Deferred/index.html#val-forever) instead of handwritten `let rec go`